File: Open3D/05_muestreo_outliers/python/main.py
import open3d as o3d
pc=o3d.io.read_point_cloud("input.ply"); down=pc.voxel_down_sample(0.02); clean,idx=down.remove_statistical_outlier(nb_neighbors=30,std_ratio=2.0); print(len(pc.points),len(down.points),len(clean.points)); o3d.io.write_point_cloud("clean.ply",clean)

File: Open3D/05_muestreo_outliers/cpp/main.cpp
#include <open3d/Open3D.h>
#include <iostream>
int main(){ auto pc=open3d::io::CreatePointCloudFromFile("input.ply"); if(!pc) return 1; auto down=pc->VoxelDownSample(0.02); auto result=down->RemoveStatisticalOutliers(30,2.0); auto clean=std::get<0>(result); std::cout<<pc->points_.size()<<" -> "<<clean->points_.size()<<"\n"; open3d::io::WritePointCloud("clean.ply",*clean); }
